automated-testing/integration-testing/readme.md

```diff
@@ -1,7 +1,82 @@
+
+## Applying Integration Testing
+
+Prior to writing integration tests the user must identify the different components of the system and their intended behaviors and inputs and outputs. The engineer must fully understand the architecture of the project.
+
+There are two main techniques for integration testing.
+
+### Big Bang
+
+Big Bang integration testing is when all components are tested as a single unit. This is best for small system as a system too large may be difficult to localize for potential errors from failed tests. This approach also requires all components in the system under test to be completed which may delay when testing begins.
+
+![Big Bang Integration Testing](./images/bigBang.jpg)
+
+### Incremental Testing
+
+Incremental testing is when two or more components that are logically related are tested as a unit. After testing the unit, additional components are combined and tested all together. This process repeats until all necessary components are tested.
+
+#### Top Down
+
+Top down testing is when higher level components are tested following the control flow of a software system. In the scenario, what is commonly refered to as stubs are used to emulate the behavior of lower level modules not yet complete or merged in the integration test.
+
+![Top Down Integration Testing](./images/topDown.png)
+
+#### Bottom Up
+
+Bottom up testing is when lower level modules are tested together. In the scenario, what is commonly refered to as drivers are used to emulate the behavior of higher level modules not yet complete or included in the integration test.
+
+![Bottom Up Integration Testing](./images/bottomUp.jpg)
+
+A third approach known as the sandwich or hybrid model combines the bottom up and town down approaches to test lower and higher level components at the same time.
+
+### Things to Avoid
+
+There is a tradeoff a developer must make between integration test code coverage and engineering cycles. With mock dependencies, test data, and multiple environments at test, too many integration tests are infeasible to maintain and become increasingly less meaningful. Too much mocking will slow down the test suite, make scaling difficult, and may be a sign the developer should consider other tests for the scenario such as acceptance or E2E.
+
+Integration tests of complex systems require high maintenance. Avoid testing business logic in integration tests by keeping test suites separate. Do not test beyond the acceptance criteria of the task and be sure to clean up any resources created for a given test. Additionally, avoid writing tests in a production environment. Instead write them in a scaled-down copy environment.
+
+## Integration Testing Frameworks and Tools
+
+Many tools and frameworks can be used to write both unit and integration tests. The following covers tools for integration tests.
```

Comment:
I know for unit testing we decided not to have a list of test frameworks and instead opted to link out to the wikipedia page with a list of all test frameworks; maybe we can focus this area on tooling that specifically will help integration testing instead of general testing tools?

Reply:
Added a few automated tooling resources and removed everything else.